androidApp/src/main/java/com/example/tasky/android/util/ViewModelFactory.kt

```diff
@@ -0,0 +1,9 @@
+package com.example.tasky.android.util
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+
+inline fun <VM : ViewModel> viewModelFactory(crossinline f: () -> VM) =
+    object : ViewModelProvider.Factory {
+        override fun <T : ViewModel> create(modelClass: Class<T>): T = f() as T
+    }
```

Comment:
The use case for defining your own `ViewModelFactory` usually is to shift to responsibility of providing dependencies to the factory itself. It would fetch the repository your viewmodel needs (this could be created in your App class and fetched from there) and pass it as an argument.